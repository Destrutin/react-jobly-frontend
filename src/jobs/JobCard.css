.JobCard-card {
    width: 300px;
    background-color: #e6e6fa;
    border-radius: 10px;
    border: 1px solid #9370db;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    text-decoration: none;
    color: inherit;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
  }
  
  .JobCard-body {
    padding: 20px;
  }
  
  .JobCard-button {
    align-self: flex-end;
    margin-top: auto;
    padding: 10px 20px;
    border: none;
    background-color: #007bff;
    color: #ffffff;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s ease;
  }
  
  .JobCard-button:hover {
    background-color: #0056b3;
  }
  