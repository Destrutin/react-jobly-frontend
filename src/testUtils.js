import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "omgsecret",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
