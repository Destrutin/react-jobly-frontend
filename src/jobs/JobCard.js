import React, {useContext, useEffect, useState} from "react";
import UserContext from "../auth/UserContext";
import "./JobCard.css"

function JobCard({id, title, salary, equity, companyName}) {

    const {hasAppliedToJob, applyToJob} = useContext(UserContext);
    const [applied, setApplied] = useState();

    useEffect(() => {
        setApplied(hasAppliedToJob(id));
    }, [id, hasAppliedToJob]);

    async function handleApply(e) {
        if (hasAppliedToJob(id)) return;
        applyToJob(id);
        setApplied(true);
    }
    return (
        <div className="JobCard-card">
            <div className="JobCard-body">
                <h4>{title}</h4>
                <p>{companyName}</p>
                <p>Salary: {salary}</p>
                <p>Equity: {equity}</p>
                <button
                    className="apply-button"
                    onClick={handleApply}
                    disabled={applied}
                >{applied ? "Applied" : "Apply"}</button>
            </div>
        </div>
    )
}

export default JobCard;