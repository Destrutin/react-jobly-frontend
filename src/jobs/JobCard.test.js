import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {UserProvider} from "../testUtils";
import JobCard from './JobCard';

test('renders company card without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
            <JobCard
                id="1"
                title="test"
                salary="test"
                equity="test"
                companyName="test company"
            />
            </UserProvider>
        </MemoryRouter>
    )
})

test('matches snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
            <JobCard
                id="1"
                title="test"
                salary="test"
                equity="test"
                companyName="test company"
            />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})