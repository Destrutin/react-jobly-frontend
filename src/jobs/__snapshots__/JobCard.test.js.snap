// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard-card"
  >
    <div
      class="JobCard-body"
    >
      <h4>
        test
      </h4>
      <p>
        test company
      </p>
      <p>
        Salary: test
      </p>
      <p>
        Equity: test
      </p>
      <button
        class="apply-button"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
